rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -220,6 +223,24 @@ impl WasmGrpcProvider {
         Ok(gas_estimate)
     }
 
+    /// Fetches balance for a given `address` and `denom`
+    pub async fn get_balance(&self, address: String, denom: String) -> ChainResult<u64> {
```

Comment:
I think we may need to make this return a U256 - it's really not clear but it seems like the `Coin.amount` spec allows it to be up to 256 bits

e.g. the Coin struct which mentions `Int`:
```
// @generated
/// Coin defines a token with a denomination and an amount.
///
/// NOTE: The amount field is an Int which implements the custom method
/// signatures required by gogoproto.
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct Coin {
    #[prost(string, tag = "1")]
    pub denom: ::prost::alloc::string::String,
    #[prost(string, tag = "2")]
    pub amount: ::prost::alloc::string::String,
}
```

and if you look in the SDK definition for Coin https://github.com/cosmos/cosmos-sdk/blob/main/types/coin.go#L19 it refers to sdk.Int

Which can be found here and has 256 bits
https://github.com/cosmos/cosmos-sdk/blob/main/math/int.go#L15